### DISCUSSION TITLE
최근 제일 관심이 가는 라이브러리 or 프레임워크

### DISCUSSION THREAD

**Discussion by michoball** [wanted-9th-3team/frontend-study#97] [최신 기술 및 산업 질문]
요즘 공부하고 있거나 사용해보니 좋았던 라이브러리나 프레임워크가 있다면 소개해주세요~

**Comment by michoball** [comment]
최근 next js 13버전을 알아보고 있습니다.  관련해서 몇가지 간단한 프로젝트도 진행을 해본 결과 많은 장점들을 볼 수 있었습니다. 

next js 13 에서는 App 폴더안에 작성한 폴더를 route로 사용할 수 있게 해주고 많은 파일 네이밍 컨벤션이 있어서 개발할 때 고민하는 파일 구조의 많은 부분을 생각하지 않게 해주었습니다. 

또한 ssr 을 할 수 있게 해주기 때문에 route.ts 파일를 사용해 server의 api 통신을 자체 구현할 수 있게 해주는 route handler를 제공합니다. 

이미지도 따로  Image 태그를 사용하면 최적화도 해주고 기본적인 navigation 관련 기능이나 search params 관련 기능을 내장하고 있어서 편리하게 사용할 수 있습니다. 

각 페이지 별 layout.tsx 파일로 레이아웃 뿐아니라 메타 태크나 지정 폰트를 정할 수도 있고, error.tsx 폴더에 에러 발생시 보여줄 view를 구현할 수도 있습니다. 

이외에도 여러 장점을 가지고 있어서 ssr로 프로젝트를 구현하기를 원한다면 nextjs 는 좋은 선택이라는 생각을 했습니다. 

하지만 ssr에 대한 이해가 충분히 뒷받침해주지 않는다면 사용하면서 ssr의 장점을 활용하지 못할 가능성이 커보인다고 생각했습니다. 

단순 react를 할 줄 안다고 잘 사용할 수 있는 것은 아닌 것 같아서 next js 도 공부가 많이 필요해보였습니다.

**Comment by javascriptbiggosoo** [comment]
주류였던 리덕스의 대안이 많이 나오는 가운데 리액트 state의 작동 방식과 동일한 recoil을 애용했는데 업데이트가 밀리고 Next에서 호환문제가 생기는 와중에 jotai가 새롭게 떠오르고 있습니다.

**Comment by WooGie911** [comment]
react-query 와 recoil 을 함께 사용해보고 있습니다.

그동안 redux-toolkit 과 redux-thunk 를 사용하여 전역관리와 리패칭을 처리하였는데 방대한 보일러 플레이트라는 가장 큰 단점을 보완하기 위해 react-query 와 recoil 을 사용해보고 있습니다. 

그동안 이론만 공부하고 실제로 사용해보지 않았는데 사이드 프로젝트를 통해 직접 코드에 적용해보니 알고 있다고 생각했던 제가 부끄럽다는 생각이 들었습니다. 

좀더 능숙하게 사용하게 노력할 예정입니다.
